provider/kubernetes/kubernetes.go

```diff
@@ -312,6 +314,15 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 							protocol = "https"
 						}
 
+						switch getStringValue(i.Annotations, annotationKubernetesProtocol, protocol) {
+						case allowedProtocolHTTPS:
+							protocol = allowedProtocolHTTPS
+						case allowedProtocolH2C:
+							protocol = allowedProtocolH2C
+						default:
+							protocol = label.DefaultProtocol
```

Comment:
Assuming we hit line 314 and have protocol be `https`: Aren't we overwriting the value to `http` again if the annotation is missing or empty (as the default of `http` will be returned to `getStringValue`, thereby causing our _default_ switch case to apply)?